Scheduler recurrenceException: edit description and the Appointment Types article (#1510)
DevExpress/devextreme-documentation: concepts/05 Widgets/Scheduler/030 Appointments/015 Appointment Types/020 All-Day Appointments.md

An all-day appointment occupies the entire day.

![Scheduler All-Day Appointment](/images/UiWidgets/Scheduler_AllDayAppointment.png)

> To make an appointment all-day, set its [allDay](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#allDay) field to **true**. An all-day appointment must have a [startDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#startDate) field. You can also set an [endDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#endDate) field if the appointment occupies more than one entire day. Time values in these fields are ignored.

Reviewer: 1) `To make an appointment all-day`...

In this case, it sounds like there is an existing appointment, and we're making this appointment an "all-day" appointment. Is this correct? Or, are we creating a new "all-day" appointment? For example:

"To create an all-day appointment, set its..."

2) Please change "one entire day" to "one day".

```suggestion
To create an all-day appointment, set its [allDay](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#allDay) field to **true**. An all-day appointment must have a [startDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#startDate) field. You can also set an [endDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#endDate) field if the appointment occupies more than one day. Time values in these fields are ignored.
```

> An all-day appointment is an appointment that occurs during the whole day.

Reviewer 2: You can't say that an appointment *occurs during* the whole day, because this would mean that an appointment just happens at a certain point of the day. while the day lasts. This doesn't say that the appointment lasts for the whole day.

```suggestion
An all-day appointment occupies the entire day.
```

![Scheduler All-Day Appointment](/images/UiWidgets/Scheduler_AllDayAppointment.png)

> To create such an appointment, set its [allDay](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#allDay) field to **true** and specify the start day in the [startDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#startDate) field; the time value is optional: 

Reviewer 3: Are we talking about appointment's data object? We should reflect this when talking about fields.

Could you please also explain, what does the following sentence mean: 'the time value is optional'?

Author: The time value of startDate and endDate are optional. Have corrected this paragraph.

> To make an appointment all-day in code, set its [allDay](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#allDay) field to **true**; the time values of [startDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#startDate) and [endDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#endDate) are ignored: 

Reviewer 2: Nothing about the fact that the startDate is required while the endDate is optional.

```suggestion
To make an appointment all-day, set its [allDay](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#allDay) field to **true**. An all-day appointment must have a [startDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#startDate) field. You can also set an [endDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#endDate) field if the appointment occupies more than one entire day. Time values in these fields are ignored.
```

    <!--JavaScript-->
    var allDayAppointment = [{
        text: "Concert",
        startDate: new Date("2021-07-27T16:00:00.000Z"),
        allDay: true
    }];

> [note]If you set the optional [endDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#endDate) field to a date after the start date, the appointment will last for more than one whole days.

Reviewer 3: ```suggestion
[note]If you set the optional [endDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#endDate) field to a date after the start date, the appointment will last for more than one whole day.
```

> [note]If you set the optional [endDate](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/#endDate) field to a date after the start date, the appointment will last for more than one whole day.

Reviewer 2: ```suggestion

```

> If your appointment data objects already have a field that acts as **allDay**, specify it in the [allDayExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#allDayExpr) option of the **Scheduler**:

Reviewer 3: I would rewrite the first part of the sentence. I'm not sure how exactly, but it is not really clear how a field can act as **allDay**. Maybe something like this:

If appointment data objects have a field that indicate whether the appointment lasts the whole day, specify this field in...

> If your appointment data objects already have a field that performs the functions of **allDay**, specify its name in the [allDayExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#allDayExpr) option of the **Scheduler**

Reviewer 2: "Have" is a weak verb, prefer using strong verbs.

```suggestion
If your appointment data objects contain a different field that performs the functions of **allDay**, specify its name in the **Scheduler**'s [allDayExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#allDayExpr) option.
```

---

To mark an appointment as all-day in the UI, toggle the *"All day"* switcher on the appointment details form. This form appears when a user [adds](/concepts/05%20Widgets/Scheduler/030%20Appointments/020%20Add%20Appointments '/Documentation/Guide/Widgets/Scheduler/Appointments/Add_Appointments/') or [updates](/concepts/05%20Widgets/Scheduler/030%20Appointments/030%20Update%20Appointments '/Documentation/Guide/Widgets/Scheduler/Appointments/Update_Appointments/') an appointment. 

> ![Scheduler All Day Appointment Details](/images/UiWidgets/Scheduler_AllDayAppointment_Details.png)

Reviewer: Since I'm unable to add a comment to line 128 (below), can you please make the following changes?

1) `All-day appointments are displayed in the All-day panel, which is visible by default.`
I'd like to rewrite this sentence without using "by default". What do you think about the following?

`All-day appointments in the All-day panel are visible (the default setting).`

2) `If you are not using such appointments, hide the All-day panel along with all-day appointments by assigning **false** to the [showAllDayPanel](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/showAllDayPanel.md '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#showAllDayPanel') option.`

I don't think the first part of the sentence is necessary, because I it's assumed that you would want to hide appointments if you're not using them, correct? What do you think about the following suggestion?

`To hide the All-day panel along with all-day appointments, set the [showAllDayPanel](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/showAllDayPanel.md '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#showAllDayPanel') option to **false**.`

Reviewer 2: Hi Albert,

> All-day appointments are displayed in the All-day panel, which is visible by default.

The phrase "which is visible by default" is about the panel, not the appointments. The whole paragraph is supposed to describe how to hide the all-day panel when there are no all-day appointments. How about the following version?

`All-day appointments are displayed in the All-day panel. If you do not use such appointments, disable the [showAllDayPanel](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/showAllDayPanel.md '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#showAllDayPanel') property to hide the All-day panel.`

Reviewer: Hi Roman. Thank you for the clarification. Please apply your suggestion.
